packages/libs/src/sdk/middleware/addRequestSignatureMiddleware.ts

```diff
@@ -24,57 +19,81 @@ export const addRequestSignatureMiddleware = ({
 }: {
   services: Pick<ServicesContainer, 'auth' | 'logger'>
 }): Middleware => {
+  let message: string | null = null
```

Comment:
Important changes:
* the cached values are _per instance_ of the middleware, since we actually have more than one active at a time now.
* The reading/updating of the values can only be done by a single request at a time. All others will block on the promise and just use the result afterwards. Same effect we wanted before (only update it every 60 seconds or when the address changes), but no more race conditions due to the multiple `await` statements involved in checking the wallet address and generating a signature.

Reply:
ok yeah this makes a ton of sense to me. hella awkward.

instead of storing the singleton of signaturePromise, we could just store a `isSignatureRefreshing` bool or something like that and just block on that value until it's false, right?

either way this seems fine/good to me! nice catches on this!